toolkit/scripts/toolchain/download_toolchain_rpm.sh

```diff
@@ -139,6 +158,32 @@ function download() {
     exit 1
 }
 
+function validate_signatures() {
+    if [ -z "$work_dir" ]; then
+        work_dir=$(mktemp -d)
+        function cleanup() {
+            rm -rf "$work_dir"
+        }
+        trap cleanup EXIT
+    fi
+
+    echo "Validating toolchain RPM: $rpm_name" | tee -a "$log_file"
+
+    for key in $allowable_gpg_keys; do
+        echo "Adding key ($key) to empty workdir at ($work_dir)" >> "$log_file"
+        rpmkeys --root "$work_dir" --import "$key" >> "$log_file"
+    done
+
+    if ! rpmkeys --root "$work_dir" --checksig --verbose "$dst_file" -D "%_pkgverify_level signature" >> "$log_file"; then
+        echo "Failed to validate toolchain package $rpm_name signature, aborting." | tee -a "$log_file"
+        exit 1
```

Comment:
I'd suggest for functions to use `return` as much as possible. This way you can leave it up to the caller to decide, if they want to exit immediately, ignore the error, or do some more processing before they are done. Combining that with `set -e`, still allows us to add no additional code, if we want to immediately exit after `return <no-zero code>`.